src/units/abilities.cpp

```diff
@@ -600,28 +614,31 @@ namespace {
 bool attack_type::get_special_bool(const std::string& special, bool simple_check) const
 {
 	{
-		std::vector<const config*> list;
-		if ( get_special_children(list, specials_, special, simple_check) ) {
+		std::vector<special_match> special_tag_matches;
+		std::vector<special_match> special_id_matches;
+		if ( get_special_children(special_tag_matches, special_id_matches, specials_, special, simple_check) ) {
 			return true;
 		}
 		// If we make it to here, then either list.empty() or !simple_check.
 		// So if the list is not empty, then this is not a simple check and
 		// we need to check each special in the list to see if any are active.
-		for(const config* entry : list) {
-			if ( special_active(*entry, AFFECT_SELF) ) {
+		for(const special_match entry : special_tag_matches) {
+			if ( special_active(entry.cfg, AFFECT_SELF, entry.tag_name) ) {
 				return true;
 			}
 		}
 	}
+
 	// Skip checking the opponent's attack?
 	if ( simple_check || !other_attack_ ) {
 		return false;
 	}
 
-	std::vector<const config*> list;
-	get_special_children(list, other_attack_->specials_, special);
-	for(const config* entry : list) {
-		if ( other_attack_->special_active(*entry, AFFECT_OTHER) ) {
+	std::vector<special_match> special_tag_matches;
+	std::vector<special_match> special_id_matches;
+	get_special_children(special_tag_matches, special_id_matches, other_attack_->specials_, special);
+	for(const special_match entry : special_tag_matches) {
```

Comment:
The last two comments (const reference and iterate over id matches) apply here too.